pandas/core/arrays/_mixins.py

```diff
@@ -157,12 +164,26 @@ def _concat_same_type(
         return to_concat[0]._from_backing_data(new_values)  # type: ignore[arg-type]
 
     @doc(ExtensionArray.searchsorted)
-    def searchsorted(self, value, side="left", sorter=None):
-        value = self._validate_searchsorted_value(value)
-        return self._ndarray.searchsorted(value, side=side, sorter=sorter)
-
-    def _validate_searchsorted_value(self, value):
-        return value
+    def searchsorted(
+        self,
+        value: ArrayLike | object,
+        side: Literal["left", "right"] = "left",
+        sorter: NumpySorter = None,
+    ) -> np.ndarray:
+        npvalue = self._validate_searchsorted_value(value)
+        return self._ndarray.searchsorted(npvalue, side=side, sorter=sorter)
+
+    def _validate_searchsorted_value(
+        self, value: ArrayLike | object
+    ) -> np.ndarray | PythonScalar:
+        if isinstance(value, ExtensionArray):
+            return value.to_numpy()
+        elif isinstance(value, np.ndarray):
+            return value
```

Comment:
this condition and the following condition both return `value`, the cast is a noop at runtime. can this be simplified?

Reply:
Yes, fixed in next commit